comms/core/src/connectivity/manager.rs

```diff
@@ -417,18 +447,50 @@ impl ConnectivityManagerActor {
 
         self.clean_connection_pool();
         if self.config.is_connection_reaping_enabled {
-            self.reap_inactive_connections(ticker_id).await;
+            self.reap_inactive_connections(task_id).await;
         }
         if let Some(threshold) = self.config.maintain_n_closest_connections_only {
-            self.maintain_n_closest_peer_connections_only(threshold, ticker_id)
-                .await;
+            self.maintain_n_closest_peer_connections_only(threshold, task_id).await;
         }
         self.update_connectivity_status();
         self.update_connectivity_metrics();
         Ok(())
     }
 
+    async fn delete_stale_peers_from_db(&mut self, task_id: u64) {
+        let start = Instant::now();
+        match tokio::time::timeout(
+            STALE_PEER_DELETE_TIMEOUT,
+            self.peer_manager.delete_all_stale_peers(self.node_identity.node_id()),
+        )
+        .await
+        {
+            Ok(res) => match res {
+                Ok(deleted) => {
+                    let len = deleted.len();
+                    if len > 0 {
+                        for node_id in deleted {
+                            self.pool.remove(&node_id);
+                        }
+                        debug!(
```

Comment:
This should probably be a warn because it is unexpected. We should not be connected to any stale peers